### PR TITLE
[civ2][1] move small & python tests to a separated pipeline

### PR DESCRIPTION
**Pull request by can-anyscale**
Split small & python tests into 2 separated pipelines, one for flaky tests and one for stable tests. The stable test step will be eventually moved into a blocking pipeline.

Test strategies:
- CI (validate that the flaky test build & stable test build runs the right number of tests)

### CODE REVIEW

File: .buildkite/pipeline.build.yml [hunk 1]
```diff
@@ -184,7 +184,7 @@
     - ./ci/env/env_info.sh
     - pip install ray[client]
     - bazel test --config=ci $(./ci/run/bazel_export_options)
-      --test_tag_filters=client_tests,small_size_python_tests
+      --test_tag_filters=client_tests,small_size_python_tests,-ci_blocking
```
Comment (aslonnie) [inline .buildkite/pipeline.build.yml:187]: can we make blocking to be the default? and add a "civ1" tag for nonblocking?

Comment (can-anyscale) [reply .buildkite/pipeline.build.yml:187]: totally

File: .buildkite/pipeline.build.yml [hunk 2]
```diff
@@ -183,9 +183,9 @@
     - bash ./ci/ci.sh prepare_docker
     - ./ci/env/env_info.sh
     - pip install ray[client]
-    - bazel test --config=ci $(./ci/run/bazel_export_options)
-      --test_tag_filters=client_tests,small_size_python_tests
-      -- python/ray/tests/...
+    - bazel test --config=ci $(./ci/run/bazel_export_options) 
+      --test_tag_filters=client_tests,small_size_python_tests 
+      $(bazel query "tests(python/ray/tests/...) intersect attr(tags, civ1, //...)")
```
Comment (aslonnie) [inline .buildkite/pipeline.build.yml:188]: why not just add civ1 into `--test_tag_filters`?

Comment (aslonnie) [reply .buildkite/pipeline.build.yml:188]: I think you want to do:

`$(bazel query "attr(tags, civ1, tests(python/ray/tests/...))")`

you do not want to use `//...` which will parse all the BUILD files, and very likely will fail.

**Review by aslonnie — COMMENTED** [review]
could you ask some one from core team that is more familiar with the tests to also have a look? on if the change makes sense?

File: .buildkite/pipeline.build.yml (hunk 2 above)
Comment (can-anyscale) [reply .buildkite/pipeline.build.yml:188]: > why not just add civ1 into --test_tag_filters

the set of tag is combined with OR, not AND, ugg (https://bazel.build/docs/user-manual#test-tag-filters)

**Comment by can-anyscale** [comment]
@rickyyx let me know if you spot anything crazy I'm doing here ,thankks

File: .buildkite/pipeline.build_core.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  conditions: ["RAY_CI_PYTHON_AFFECTED"]
+  instance_size: medium
+  commands:
+    - cleanup() { if [ "${BUILDKITE_PULL_REQUEST}" = "false" ]; then ./ci/build/upload_build_info.sh; fi }; trap cleanup EXIT
+    - DL=1 ./ci/env/install-dependencies.sh
+    - bash ./ci/ci.sh prepare_docker
+    - ./ci/env/env_info.sh
+    - pip install ray[client]
+    - bazel test --config=ci $(./ci/run/bazel_export_options) 
+      --test_tag_filters=client_tests,small_size_python_tests,-civ1
```
Comment (rickyyx) [inline .buildkite/pipeline.build_core.yml:13]: How are we deciding which tests to be included in v2 here?

**Review by rickyyx — COMMENTED** [review]
How are we deciding

**Review by rickyyx — APPROVED** [review]
Excited to see this happening!

**Comment by rickyyx** [comment]
> Excited to see this happening!

Let's also let folks know when this is going to happen. Should we maybe send it to #open-source?

**Comment by can-anyscale** [comment]
@rickyyx: I will yes. Will prepare the one-pager for core team to understand the process as well when this becomes blocking. Currently it is not blocking yet so the semantic is still exactly the same.

**Pull request merged**